### PR TITLE
MINOR: Remove redundant `if` condition.

### PR DESCRIPTION
**Pull request by kamalcph**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by kamalcph** [comment]
@bbejeck 
Could you please review this minor PR?

**Comment by kamalcph** [comment]
@bbejeck @mjsax 
Please take a look when you have time.

**Comment by mjsax** [comment]
I think the PR makes sense. Is there any danger to make the code less readable? \cc @guozhangwang @vvcephei

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java [hunk 1]
```diff
@@ -261,15 +260,11 @@ private int sumCapacity(final Collection<ClientState> values) {
...
 
         boolean hasNewPair(final TaskId task1,
-                           final Set<TaskId> taskIds,
-                           final boolean active) {
+                           final Set<TaskId> taskIds) {
             if (pairs.size() == maxPairs) {
                 return false;
             }
             for (final TaskId taskId : taskIds) {
-                if (!active && !pairs.contains(pair(task1, taskId))) {
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java:270]: I think this is a good catch. For checking `hasNewPair` the logic for avoiding conflicts on either active-active or active-standby should be the same. cc @bbejeck for another look.

Comment (bbejeck) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignor.java:270]: Yes, this is a good catch. We used to need the `active` check before #5390, but I missed that during the refactoring.  We can remove the redundant `if` condition.

**Review by bbejeck — APPROVED** [review]
Thanks for the patch @kamalcph LGTM.

**Pull request merged**